algorithms/Java/sorting/cyclic-sort.java

```diff
@@ -0,0 +1,29 @@
+package CyclicSort;
+import java.util.Arrays;
+
+// when given nos. range from 1 - n then use cyclic sort..
+public class CyclicSort {
+    public static void main(String[] args) {
+        int[] arr ={4,2,5,3,1};
```

Comment:
Hi @Ashwin-exe 
Thanks for your effort,

Please fix the following:
* The algorithm stuck in this case: **int[] arr ={4,2,5,9};**
with the following error msg: **java.lang.ArrayIndexOutOfBoundsException: Index 8 out of bounds for length 4**

Thanks.

Reply:
Hey!! @TawfikYasser, I **fixed it** thanks for mentioning the issue 😊